### DISCUSSION TITLE
What is github

### DISCUSSION THREAD

**Discussion by renning100** [community/community#62011] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

Please teach me about github

**Comment by TrendSpider001** [comment, marked as answer]
GitHub is a code hosting platform for collaboration and version control.

GitHub lets you (and others) work together on projects.
Repository
A GitHub repository can be used to store a development project.

It can contain folders and any type of files (HTML, CSS, JavaScript, Documents, Data, Images).

A GitHub repository should also include a licence file and a README file about the project.

A GitHub repository can also be used to store ideas, or any resources that you want to share.

Branch
A GitHub branch is used to work with different versions of a repository at the same time.

By default a repository has a master branch (a production branch).

Any other branch is a copy of the master branch (as it was at a point in time).

New Branches are for bug fixes and feature work separate from the master branch. When changes are ready, they can be merged into the master branch. If you make changes to the master branch while working on a new branch, these updates can be pulled in.
Commits
At GitHub, changes are called commits.

Each commit (change) has a description explaining why a change was made.

Pull Requests
Pull Requests are the heart of GitHub collaboration.

With a pull request you are proposing that your changes should be merged (pulled in) with the master.

Pull requests show content differences, changes, additions, and subtractions in colors (green and red).

As soon as you have a commit, you can open a pull request and start a discussion, even before the code is finished.

**Reply by renning100** [reply]
cool
